libr/debug/p/debug_native.c

```diff
@@ -1091,9 +1091,7 @@ static RList *r_debug_native_map_get(RDebug *dbg) {
 		}
 #if __KFBSD__
 		// 0x8070000 0x8072000 2 0 0xc1fde948 rw- 1 0 0x2180 COW NC vnode /usr/bin/gcc
-		if (sscanf (line, "%s %s %d %d 0x%s %3s %d %d",
-				&region[2], &region2[2], &ign, &ign,
-				unkstr, perms, &ign, &ign) != 8) {
+		if (r_str_scanf (line, "%.s %.s %d %d 0x%.s %.s %d %d", sizeof (region[2]), &region[2], sizeof (region2[2]), &region2[2], &ign, &ign, sizeof (unkstr), unkstr, sizeof (perms) perms, &ign, &ign) != 8) {
```

Comment:
```suggestion
		if (r_str_scanf (line, "%.s %.s %d %d 0x%.s %.s %d %d", sizeof (region)-2, &region[2], sizeof (region2)-2, &region2[2], &ign, &ign, sizeof (unkstr), unkstr, sizeof (perms) perms, &ign, &ign) != 8) {
```

Reply:
same here, the whole format string and arguments were wrong.

The meaning of * in sscanf is documented in the manpage. and bascially means to skip that argument, so you dont need to pass any argument, and we can remove all the silly variables that do nothing in this code